### PR TITLE
Adding bluetooth traffic parse logic in the pairinghandler#159

### PR DESCRIPTION
**Pull request by TheBonheurs**
- Relevant Issues: #
- Related Pull Requests: -

* * *

## What
Add the possibility to send multiple types data over Bluetooth and also being able to seamlesly parse it at the receivers end.

## Why
Because we need to send multiple types of data via Bluetooth like contacts but also singleTaps as means of verification. The bluetoothHandler should be able to handle that.

## How
By adding the more specific calls to the bluetooth activity one should be able to properly recieve variying data on their device via bluetooth.

## Alternative implementation
Instance specific implementations were discussed with the team over whatsapp. Think things like a new wrapper for a list of contacts or just a list of contacts etc.

## Notes
Most of the work went into APairingHandler and it's tests so that woulde be top priority to review. For @jverbraeken please look at the new wrapper classes. For everybody, please look at the suppresswarnings statements and comment if you don't agree

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java [hunk 1]
```diff
@@ -86,21 +104,67 @@ boolean checkExists(final Contact contact) throws IOException {
...
+     * and dispatches the data approprietly in order to parse it.
      *
-     * @param bytes byte array representing a contact
-     * @return Whether or not the process finished successfully
+     * @param bytes Data represented as bytes (usually received from the socket)
      */
-    Contact saveContact(final byte[] bytes) {
-        LOGGER.info("Saving these bytes: %s", bytes);
-        final Contact contact;
+    @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "PMD.StdCyclomaticComplexity", "PMD.CyclomaticComplexity"})
```
Comment (KilianCallebaut) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java]: can you comment on these suppressions

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java [hunk 2]
```diff
@@ -86,21 +104,67 @@ boolean checkExists(final Contact contact) throws IOException {
...
+            case CONTACT_CLASS:
+                LOGGER.info("Content was a contact");
+                saveContact((Contact) dataWrapper.getData());
+                break;
+            case "ArrayList":
+                LOGGER.info("Content was a List");
+                final ArrayList<Object> list = (ArrayList<Object>) dataWrapper.getData();
+                if (!list.isEmpty()) {
+                    for (final Object o : list) {
+                        if (o.getClass().getSimpleName().equals(CONTACT_CLASS)) {
```
Comment (KilianCallebaut) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java]: can't you also just say instanceof Contact? Don't know if this is better though.

File: app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java [hunk 3]
```diff
@@ -234,11 +233,11 @@ public void stopDiscovering() {
      */
     @Subscribe
     public void onEvent(final BluetoothConnectedEvent event) {
-        try {
+        /*try {
             bluetoothHandler.sendAllContacts();
         } catch (final IOException e) {
             LOGGER.warn("Writing all contacts issued an IOexception", e);
-        }
+        }*/
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java]: Can you remove commented code?

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java [hunk 4]
```diff
@@ -14,19 +17,25 @@
...
  */
-abstract class APairingHandler implements IPairingHandler, Serializable {
+// default value is now 7, I have 9 with one probably gone soon
+@SuppressWarnings("checkstyle:classdataabstractioncoupling")
+abstract class APairingHandler implements Serializable {
 
     private static final Logger LOGGER = LoggerFactory.getLogger("APairingHandler");
     private static final long serialVersionUID = 1656974573024980860L;
-
+    private static final String CONTACT_CLASS = "Contact";
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java]: This should be in `ConstantKeywords`, right @jverbraeken?

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java [hunk 5]
```diff
@@ -36,17 +45,25 @@
...
+     * A QOL mehtod that will send a certain list of contacts
+     *
+     * @param contacts A list with contacts
      */
-    @Override
-    public void sendAllContacts() throws IOException {
-        final List<Contact> list = database.getAllContacts();
-        for (final Contact e : list) {
-            sendContact(e);
+    public void sendAllContacts(final Collection<Contact> contacts) throws IOException {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/APairingHandler.java]: Should sending the contacts be part of the PAIRING handler? Sending data isn't really a part of the pairing progress, right?

File: app/src/main/java/com/nervousfish/nervousfish/modules/pairing/IPairingHandler.java [hunk 6]
```diff
@@ -1,16 +0,0 @@
-package com.nervousfish.nervousfish.modules.pairing;
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/pairing/IPairingHandler.java]: Can you explain why the `IPairingHandler` is removed?

File: app/src/test/java/com/nervousfish/nervousfish/modules/pairing/APairingHandlerTest.java [hunk 7]
```diff
@@ -69,25 +80,56 @@ void send(byte[] buffer) {
...
 
     @Before
     public void setup() throws Exception {
-        IServiceLocator serviceLocator = mock(IServiceLocator.class);
+        serviceLocator = mock(IServiceLocator.class);
         database = mock(IDatabase.class);
         when(serviceLocator.getDatabase()).thenReturn(database);
+        constants = mock(IConstants.class);
+        when(serviceLocator.getConstants()).thenReturn(constants);
+        when(constants.getDatabaseContactsPath()).thenReturn(new File("E:\\docs\\NervousFish\\app\\src\\main\\AndroidManifest.xml").getAbsolutePath());
```
Comment (ericcornelissen) [inline app/src/test/java/com/nervousfish/nervousfish/modules/pairing/APairingHandlerTest.java]: This path is very specific and won't work on all machines. Can't you use a relative path?

**Pull request closed without merging**